### PR TITLE
✨ Implement arbitrary for ulid

### PR DESCRIPTION
**Pull request by vecerek**
Implements #4009

Adds arbitrary for `ulid`, according to the [ulid spec](https://github.com/ulid/spec).

**_Category:_**

- [x] ✨ Introduce new features
- [ ] 📝 Add or update documentation
- [ ] ✅ Add or update tests
- [ ] 🐛 Fix a bug
- [ ] 🏷️ Add or update types
- [ ] ⚡️ Improve performance
- [ ] _Other(s):_ ...
  
  

**_Potential impacts:_**

- [ ] Generated values
- [ ] Shrink values
- [ ] Performance
- [ ] Typings
- [ ] _Other(s):_ ...

### CODE REVIEW

**Review by dubzzz — COMMENTED** [review]
I just did a quick pass on the PR from my mobile and it looks good. I'll just re-read it before approving. Thank you so much for this contribution.

**Comment by vecerek** [comment]
@dubzzz would you like me to address the formatting/linting issues and the requirement to declare the version bump? I may need some guidance with the latter.

**Comment by dubzzz** [comment]
For the first one, you have to run:

```sh
yarn format
```

For the bumps (declare minor on fast-check and mark all others as unchanged):

```sh
yarn version:bump
```

**Comment by vecerek** [comment]
Thanks for the help, @dubzzz. I believe I addressed all the outstanding issues now 🙂

File: packages/fast-check/test/unit/arbitrary/_internals/mappers/UintToBase32String.spec.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import fc from 'fast-check';
+import {
+  paddedUintToBase32StringMapper,
+  uintToBase32StringUnmapper,
+} from '../../../../../src/arbitrary/_internals/mappers/UintToBase32String';
+
+describe('uintToBase32StringUnmapper', () => {
+  it('is able to unmap any mapped value', () =>
```
Comment (dubzzz) [inline packages/fast-check/test/unit/arbitrary/_internals/mappers/UintToBase32String.spec.ts:8]:
```suggestion
  it('should be able to unmap any mapped value', () =>
```

File: packages/fast-check/src/arbitrary/_internals/mappers/UintToBase32String.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+}
+
+/** @internal */
+export function uintToBase32StringMapper(num: number, constLength: number | undefined = undefined): string {
+  if (num === 0) return pad('0', constLength ?? 1);
+
+  let base32Str = '',
+    remaining = num,
+    symbolsLeft = Math.floor(getBaseLog(32, num)) + 1;
+  while (symbolsLeft > 0) {
```
Comment (dubzzz) [inline packages/fast-check/src/arbitrary/_internals/mappers/UintToBase32String.ts:90]: I'd rather go for a for-loop there

**Review by dubzzz — COMMENTED** [review]
Just two minor comments. We should also add the doc part for ulid and the no regression e2e test

**Comment by vecerek** [comment]
@dubzzz I've [generated the jest snapshot](https://github.com/dubzzz/fast-check/pull/4020/commits/8cce6abfc001beede0192cb43369785733c38b87) covering ulids for the regression tests, and added some [web documentation](https://github.com/dubzzz/fast-check/pull/4020/commits/c6382094db72cbaf8dc1313e53f0adf366b6987c), as well. Let me know what you think.

**Review by dubzzz — DISMISSED** [review]
Looks perfect. Thanks for the contribution. I'll just recheck one last time from a computer but it seems perfect

**Comment by vecerek** [comment]
@dubzzz I had to run `yarn workspace fast-check update:examples`. I wasn't aware examples were auto-generated but it seems obvious in retrospect 😄

**Review by dubzzz — APPROVED** [review]
Perfect 👍

**Comment by dubzzz** [comment]
Will be merged in a few minutes. Just waiting the tests to run. I'll try to release it soon

**Pull request merged**